omnisolver/samplers/van.py

```diff
@@ -0,0 +1,188 @@
+import dimod
+import torch
+import time
+import numpy as np
+from functools import partial
+import random
+from ...van_orig.src_hop_sk.sk import SKModel
+from ...van_orig.src_hop_sk.made2 import MADE
+
+class VanSampler(dimod.Sampler):
+    """Implementatoin of simple random-sampler."""
+
+    def __init__(self, **kwargs):
+        d = kwargs.pop('cuda')
+        self.n = kwargs.pop('n')
+        self.device = 'cuda:{}'.format(d) if d >=0 else 'cpu'
+        self.net_params = {
+            'n': self.n,
+            'net_depth': kwargs.pop('net_depth'),
+            'net_width': kwargs.pop('net_width'),
+            'bias': kwargs.pop('bias', False),
+            'z2': kwargs.pop('z2', False),
+            'res_block': kwargs.pop('res_block', False),
+            'x_hat_clip': kwargs.pop('x_hat_clip'),
+            'epsilon': kwargs.pop('epsilon'),
+            'device': self.device
+        }
+
+
+    def _prepare_sampling(self, bqm, **kwargs):
+        self.ham = ModifiedSKModel(self.n, self.device, bqm)
+        self.ham.J.requires_grad = False
+
+
+        self.net = MADE(**self.net_kwargs)
+        self.net.to(self.device)
+
+    def my_log(self, s):
```

Comment:
Again, this is probably a leftover from original VAN. What do you think about using standard logging instead of this custom solution? I think it would be much more flexible.

Reply:
Yes I kept this from the original VAN, because it is a mess and didn't want to toych this. But I agree. We should probably agree on some logging format.